[JIT] Fix python binding error with empty containers in init.cpp

ttanpcs opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82029
* #81917
* #81916
* #81837
* #81836
* __->__ #81786
* #81785

- toTypeInferredIValue will throw an error when given an empty container because it isn't able to tell what kind of container it is. Thus empty containers are ignored in addArgumentValue/s, overlaps, and is_alias_of.

Commit:
[JIT] Fix python binding error with empty containers in init.cpp

[ghstack-poisoned]

davidberard98 reviewed (commented):
could you also add some tests for these previously-failing cases?

Commit:
Update on "[JIT] Fix python binding error with empty containers in init.cpp"


- toTypeInferredIValue will throw an error when given an empty container because it isn't able to tell what kind of container it is. Thus empty containers are ignored in addArgumentValue/s, overlaps, and is_alias_of. 


[ghstack-poisoned]

Commit:
Update on "[JIT] Fix python binding error with empty containers in init.cpp"


- toTypeInferredIValue will throw an error when given an empty container because it isn't able to tell what kind of container it is. Thus empty containers are ignored in addArgumentValue/s, overlaps, and is_alias_of. 


[ghstack-poisoned]

Commit:
Update on "[JIT] Fix python binding error with empty containers in init.cpp"


- toTypeInferredIValue will throw an error when given an empty container because it isn't able to tell what kind of container it is. Thus empty containers are ignored in addArgumentValue/s, overlaps, and is_alias_of. 


[ghstack-poisoned]

goldenxuett commented:
@pytorchbot merge
